File: src/main/java/nl/rabobank/assessment/ui/rest/model/request/PersonRequest.java
package nl.rabobank.assessment.ui.rest.model.request;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import java.time.LocalDate;

public record PersonRequest(

		@NotNull
		String firstName,

		@NotNull
		String lastName,

		@NotNull
		LocalDate dateOfBirth,

		@NotNull
		String street,

		@NotNull
		Integer houseNumber,

		String houseNumberAdditions,

		@NotNull
		@Pattern(regexp = "(?i)^\\d{4}\\s?[A-Z]{2}$")
		String postalCode,

		@NotNull
		String city,

		@NotNull
		String country
) {}

File: src/main/kotlin/nl/rabobank/assessment/ui/rest/model/request/PersonRequest.kt
package nl.rabobank.assessment.ui.rest.model.request

import java.time.LocalDate
import javax.validation.constraints.Pattern

data class PersonRequest(

    val firstName: String,
    val lastName: String,
    val dateOfBirth: LocalDate,
    val street: String,
    val houseNumber: Int,
    val houseNumberAdditions: String?,
    @Pattern(regexp = "(?i)^\\d{4}\\s?[A-Z]{2}$")
    val postalCode: String,
    val city: String,
    val country: String)